src/utils/misc.ts

```diff
@@ -22,3 +26,29 @@ export const formattedDate = (timestamp: number): string => {
   const month = new Intl.DateTimeFormat('en-US', { month: 'short' }).format(date);
   return `${month.split(',')[0]} ${day}${nth(day)}, ${date.getFullYear()}`;
 };
+
+export function to32Bytes(hexString: string): string {
+  const bigInt = hexToBigInt(hexString as Address, { size: 32 });
+  const bytes32 = numberToHex(bigInt, { size: 32 });
+  return bytes32;
+}
+
+export function formatProofData({ merkle_root, nullifier_hash, proof }: ISuccessResult) {
+  // Format merkle root to bytes32
+  const merkle_root_bytes32 = to32Bytes(merkle_root);
+
+  const [decodedMerkleRoot] = decodeAbiParameters(
+    parseAbiParameters('uint256 merkle_root'),
+    merkle_root_bytes32 as Hex,
+  );
+  const [decodedNullifierHash] = decodeAbiParameters(
+    parseAbiParameters('uint256 nullifier_hash'),
+    nullifier_hash as Hex,
+  );
+  const [decodedProof] = decodeAbiParameters(parseAbiParameters('uint256[8] proof'), proof as Hex);
+  const proofData = encodePacked(
+    ['uint256', 'uint256', 'uint256[8]'],
+    [decodedMerkleRoot, decodedNullifierHash, decodedProof],
+  );
+  return proofData;
+}
```

Comment:
Why using camel case here?

Reply:
snake case*

Reply:
https://docs.worldcoin.org/reference/idkit#types due to how it comes from sdk

Reply:
[d08a1a5](https://github.com/defi-wonderland/world-id-goats-dao-ui/pull/29/commits/d08a1a5a96a0128a2f360a37e173da21c557b137)